shared-libs/transitions/src/config.js

```diff
@@ -1,13 +1,15 @@
 let config = {};
 let translations = {};
-
+let transitionsLib = {};
 module.exports = {
-  init: (sourceConfig, sourceTranslations) => {
+  init: (sourceConfig, sourceTranslations, transitions) => {
     config = sourceConfig;
     translations = sourceTranslations;
+    transitionsLib = transitions;
```

Comment:
:) Good question!  It boils down to how the create_user_for_contacts transition passes its version of the config to the `user-management` shared-lib when it goes to create the new user. The `user-management` shared-lib needs access to the transitions because it uses them to [send the token login messages](https://github.com/medic/cht-core/blob/13f0c322166db97c35e16d7de550e18c7ba5eee6/shared-libs/user-management/src/token-login.js#L218)

Feels like a good candidate for another new shared-lib, but not sure we want to do that now in this PR....

Reply:
Can't the users-management library just get an extra parameter for the messages lib directly?

Reply:
Injecting the `messages` lib like you suggested would break in the api code that needs to require `@medic/user-management` (without also having to refactor that code).  (Since those requires are happening before the transitionsLib is initialized in api, so you cannot access `config.getTransitionsLib().messages`.)

Honestly I have spent the last couple of hours trying to unravel the rat's nest that is these configs (particularly the way that sentinel has a config, but only passes the settings values to `transitions` (and not the rest of its functions) and I have not been able to find a more clean way to do this then what we already have here.

Reply:
Okay, so I ended up pouring a few more hours down this rabbit hole and came up with a [a solution](https://github.com/medic/cht-core/pull/7709/commits/d10936fa7f14ab244d3dc4df35a844422d4b9355) that seems to at least move the code base in a good direction (instead of just making the rat's nest worse...).  Basically, it makes the `transitions` config to just be a simple pass-through just like we are doing for `contacts` and `user-management`. The functionality in that commit seems good but I ended up reverting it after it became clear that I would have to go through and update pretty much every since unit/integration test in `transitions`.  Let me know if you actually like the refactor approach from the commit and I can pull it back in and fix the tests.  (Did not want to waste a bunch of time fixing tests to work with an approach we did not actually want to go with....)

Reply:
Ha, thanks for investigating! 
I agree that your proposed solution is taking the code in the right direction, I like it, it'd still leave us with some spaghetti logic to deal with later, tho. With the added layers of libs, I think maybe a bigger refactor is in order to clean things up (the more I think about it, the more I believe config and translations need to be accessible differently). 

I'm leaving this up to you.

Reply:
Seeing as how we have made it this far, I am going to go ahead and make the change!  I agree with you that more refactoring is needed t have a better way of dealing with the config/translations, but this is a step in the right direction and should make any eventual refactors even easier.